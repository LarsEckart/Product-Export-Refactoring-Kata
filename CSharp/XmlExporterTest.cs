using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace export
{
    [UseReporter(typeof(DiffReporter))]
    [TestClass]
    public class XmlExporterTest
    {

        [TestMethod]
        public void ExportFull()
        {
            var orders = new List<Order>
            {
                SampleModelObjects.RecentOrder
            };
            Approvals.Verify(XmlExporter.ExportFull(orders));
        }

        [TestMethod]
        public void ExportHistory()
        {
            var orders = new List<Order>
            {
                SampleModelObjects.RecentOrder
            };
            Approvals.Verify(XmlExporter.ExportHistory(orders));
        }

        [TestMethod]
        public void ExportTaxDetails()
        {
            var orders = new List<Order>
            {
                SampleModelObjects.RecentOrder,
                new Order("1234", Util.FromIsoDate("2017-09-01T00:00Z"),
                    SampleModelObjects.FlagshipStore, 
                    new Product[] { SampleModelObjects.CherryBloom, SampleModelObjects.Makeover })
                
            };
            Approvals.Verify(XmlExporter.ExportTaxDetails(orders));
        }

        [TestMethod]
        public void ExportStore()
        {
            Approvals.Verify(XmlExporter.ExportStore(SampleModelObjects.BigStore));
        }


    }


}