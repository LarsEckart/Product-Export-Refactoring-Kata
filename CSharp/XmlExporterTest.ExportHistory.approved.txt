<?xml version="1.0" encoding="utf-16"?>
<orderHistory>
  <order date="2018-09-01T00:00:00Z" totalDollars="0">
    <product id="LIPSTICK01">Cherry Bloom</product>
  </order>
</orderHistory>